pkg/kafka/request.go

```diff
@@ -99,7 +173,10 @@ func ReadRequest(reader io.Reader) (*RequestMessage, error) {
 		return nil, err
 	}
 
-	req.version = req.getVersion()
+	if len(req.rawMsg) < 12 {
+		return nil, fmt.Errorf("unable to read full request")
```

Comment:
This needs a more descriptive error message, something like:
`unexpected end of request (length < 12 bytes)`

Reply:
done